### PR TITLE
Feature/win debug dll (closes #1035)

### LINKED ISSUE #1035: Should we support R CMD SHLIB -d on Windows only?
Via email, a user alerted me to the fact that he cannot create a gdb-compliant file on Windows when using `sourceCpp()`.

And that appears to be the case.  The bottom of `R CMD SHLIB --help` shows

```
Windows only:
  -d, --debug           build a debug DLL
```

And seemingly (when using `sourceCpp()`) one cannot get by via `~/.R/Makevars`.   Looking at `R/Attributes.R` it appears that we'd "just" need a new argument to `sourceCpp()` and insert an `ifelse()` line around line 124 just as we do to support `--preclean` and `--dry-run`.   

Shall we do that? I guess there is no real reason not to...

### PR DESCRIPTION
**Pull request by eddelbuettel**
See the discussion and resolution in #1035 -- this addresses a narrow need for debugging when using windows and `sourceCpp()`.   

#### Checklist

- [x] Code compiles correctly
- [x] `R CMD check` still passes all tests
- [x] Document the changes by file in [ChangeLog](https://github.com/RcppCore/Rcpp/blob/master/ChangeLog)
- [x] User confirmed issue as addressed

### CODE REVIEW

File: R/Attributes.R [hunk 1]
```diff
@@ -62,7 +63,12 @@ sourceCpp <- function(file = "",
         if (grepl(' ', basename(file), fixed=TRUE)) {
             stop("The filename '", basename(file), "' contains spaces. This ",
                  "is not permitted.")
-        }                                                       # #nocov end
+        }
+    } else {
+        if (verbose && windowsDebugDLL) {
+            message("The 'windowsDebugDLL' toggle is ignored on non-Windows platforms.")
+            windowsDebugDLL <- FALSE    # now we do not need to deal with OS choice below
```
Comment (coatless) [inline R/Attributes.R]: Shouldn't this be outside the if()? As it stands now, this will only trigger if `verbose = TRUE` is passed (it normally isn't.)

Comment (eddelbuettel) [reply R/Attributes.R]: Nice catch.  I think you are correct.  Will fix.

Comment (eddelbuettel) [reply R/Attributes.R]: New variant:

```R
        if (windowsDebugDLL) {                                                                                                                                                                                     
            if (verbose) message("The 'windowsDebugDLL' toggle is ignored on non-Windows platforms.")                                                                                                              
            windowsDebugDLL <- FALSE    # now we do not need to deal with OS choice below                                                                                                                          
        }                                                       # #nocov end                                                                                                                                       
    }       
```

Better...

File: inst/NEWS.Rd [hunk 2]
```diff
@@ -9,6 +9,8 @@
     \itemize{
       \item Safer \code{Rcpp_list*}, \code{Rcpp_lang*} and
       \code{Function.operator()} (Romain in \ghpr{1014}, \ghit{1015}).
+      \item A number of \code{#nocov} markers were added (Dirk in
+      \ghprP1036}).
```
Comment (coatless) [inline inst/NEWS.Rd]:
```suggestion
      \ghpr{1036}). 
```

File: R/Attributes.R [hunk 3]
```diff
@@ -62,7 +63,15 @@ sourceCpp <- function(file = "",
         if (grepl(' ', basename(file), fixed=TRUE)) {
             stop("The filename '", basename(file), "' contains spaces. This ",
                  "is not permitted.")
-        }                                                       # #nocov end
+        }
+    } else {
```
Comment (coatless) [inline R/Attributes.R]:
```suggestion
     else {
```

Comment (coatless) [reply R/Attributes.R]: This is what broke the build.

File: inst/NEWS.Rd (hunk 2 above)
Comment (eddelbuettel) [reply inst/NEWS.Rd]: Hmpf. I could swear I can a check which should spotted that.  Will fix!

File: R/Attributes.R [hunk 4]
```diff
@@ -67,7 +67,7 @@ sourceCpp <- function(file = "",
     } else {
         if (windowsDebugDLL) {
             if (verbose) {
-                message("The 'windowsDebugDLL' toggle is ignored on "
+                message("The 'windowsDebugDLL' toggle is ignored on ",
```
Comment (coatless) [inline R/Attributes.R]: Good catch on this one.

Comment (eddelbuettel) [reply R/Attributes.R]: Well it was a last-second edit to comply with the @jjallaire style of less than 80 columns. I find 100 or so easier and this is what I got for it :)

**Comment by eddelbuettel** [comment]
Thanks for the review.  There isn't all that much else here, and you caught the one obvious thinko, so I think I'll merge this now.

File: inst/NEWS.Rd [hunk 5]
```diff
@@ -17,18 +19,27 @@
       \ghit{1017}).
       \item Invisible return object are now supported via new option (Kun Ren
       in \ghpr{1025} fixing \ghit{1024}).
+      \item Unavailable packages referred to in \code{LinkingTo} are now
+      reported (Dirk in \ghpr{1027} fixing \ghit{1026}).
+      \item The \code{sourceCpp} function can now create a debug DLL on
+      Windows (Dirk in \ghpr{1037} fixing \ghit{ghit}).
```
Comment (coatless) [inline inst/NEWS.Rd]: Also, there's `\ghit{ghit}`

Comment (eddelbuettel) [reply inst/NEWS.Rd]: Bah.  Something must be wrong with the water.  Or the coffee.  Or both.

Changed to `\ghit{1035}`. Thanks again.

**Pull request merged**